shell/shared/db.js

```diff
@@ -86,6 +86,46 @@ Grains = new Mongo.Collection("grains");
 //   publicId:  An id used to publicly identify this grain. Used e.g. to route incoming e-mail and
 //       web publishing. This field is initialized when first requested by the app.
 
+RoleAssignments = new Mongo.Collection("roleAssignments");
+// Edges in the permissions sharing graph.
+//
+// To share a permission with another user is to declare "if I have this permission, then so should
+// the recipient". A bundle of such declarations by a single sharer directed at a single recipient
+// for a single grain is called a "role assignment". A grain's owner always has every permission,
+// but permissions for other users must be computed from this collection.
+//
+// Every role assignment implicitly shares the special "can view grain" permission, which dictates
```

Comment:
OK, I see what you're getting at now -- the empty permission may actually grant the user some ability to access the grain, depending on the app. Whether this can be described as "view" depends on the app, though.

But there's another issue here: It sounds like you're saying that deleting a RoleAssignment will transitively delete downstream assignments if they aren't backed elsewhere. I actually don't think we want that -- If Alice shares to Bob who shares to Carol, and then Alice temporarily revokes Bob (perhaps by accident) and then restores his access, Carol's access needs to be restored as well. So the Bob -> Carol edge can't really be deleted. And meanwhile, we probably can't do a full access check for every capability in your account when rendering a list, so Carol will still see the grain even though she has lost access. Once she clicks on it, she'll get an error. Annoying UX but probably rare... and actually arguably less annoying then having the document disappear entirely without explanation.